VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitVideoFragment.java

```diff
@@ -213,43 +211,41 @@ public void onResume() {
     @Override
     public void setUserVisibleHint(boolean isVisibleToUser) {
         super.setUserVisibleHint(isVisibleToUser);
-        if ( ViewPagerDownloadManager.instance.inInitialPhase(unit) )
+        if (ViewPagerDownloadManager.instance.inInitialPhase(unit))
             return;
         if (isVisibleToUser) {
-            try {
-                if (playerFragment != null) {
-                    playerFragment.setCallback(this);
-                    playerFragment.handleOnResume();
-                }
-            } catch (Exception ex) {
-                logger.error(ex);
-            }
-        }else{
+            setVideoPlayerState(true, this);
+        } else {
             // fragment is no longer visible
-            try {
-                if (playerFragment != null) {
-                    playerFragment.setCallback(null);
-                    playerFragment.handleOnPause();
-                }
-            } catch (Exception ex) {
-                logger.error(ex);
-            }
+            setVideoPlayerState(false, null);
         }
     }
+
     @Override
     public void run() {
-        if ( this.isRemoving() || this.isDetached()){
+        if (this.isRemoving() || this.isDetached()) {
             ViewPagerDownloadManager.instance.done(this, false);
         } else {
-            try {
-                if (playerFragment != null) {
-                    playerFragment.setCallback(this);
-                    playerFragment.handleOnResume();
-                }
-                ViewPagerDownloadManager.instance.done(this, true);
-            } catch (Exception ex) {
-                logger.error(ex);
+            setVideoPlayerState(true, this);
+            ViewPagerDownloadManager.instance.done(this, true);
+        }
+    }
+
+    /**
+     * Sets the state of video player inside {@link PlayerFragment} to playing or paused
+     *
+     * @param playing  Boolean to set state to playing or paused
```

Comment:
The description could perhaps be changed to "True for playing the video player, false for pausing it.", or something similar for more clarity.